lib/player.rb

```diff
@@ -1,17 +1,35 @@
 module Scrabble
   class Player
-    attr_reader :name, :plays
+    attr_reader :name, :plays, :valid_word
+    attr_accessor :tiles
 
     def initialize(name)
       @name = name
       @plays = []
+      @tiles = []
+    end
+
+    def draw_tiles(tile_bag)
+      number = 7 - @tiles.length
+      @tiles += tile_bag.draw_tiles(number)
     end
 
     def play(word)
+      word = word.upcase
       if won?
         return false
       else
-        @plays.push(word)
+        word_in_array = word.split("")
+        remaining_tiles = @tiles.dup
+
+        word_in_array.each do |l|
+          remaining_tiles.slice!(remaining_tiles.index(l)) if remaining_tiles.include?(l)
```

Comment:
remaining_tiles is a local variable copy of @tiles, the tiles in the player's hand. The iterator goes through each letter in word_in_array (the word submitted by the user split into an array). The if statement checks if that letter even exists in the remaining_tiles array. This is necessary because otherwise index.("l") will return nil if the letter is not in remaining_tiles and we will get a TypeError. Once we know that 'l' is in remaining_tiles, we find its index, then the 'slice!' method will remove the element at that index from remaining_tiles! This will avoid deleting duplicates of a letter present in remaining_tiles (with delete or - methods, for example) because we are getting rid of an item at a particular INDEX. I believe '.delete_at' will do the same thing as 'slice' in this case.

If all the letters in 'word' are present in the remaining_tiles array, the sizes will match! Then we know this was a valid word submission.